example/README.md

```diff
@@ -1,88 +1,223 @@
+
+> The generator will process `lib` and `test` folders separately and create a separate database in each one, if it finds
+> annotations there. This is useful if you need a separate test DB. If you're just writing tests for your own code, you
+> won't have any annotations in the `test` folder so no DB will be created there.
+
+Creating a store
+----------------
+
+`Store` is your entrypoint to ObjectBox - see below how to open it, based on the SDK you're using.
+
+### Flutter apps
+
+On mobile devices, you should store data in your app documents directory - it stays there even when you close the app.
+
+Use `getApplicationDocumentsDirectory()` from the `path_provider` package to retrieve this directory.
 
-To configure ObjectBox properly, you can use `getApplicationDocumentsDirectory()` from the `path_provider` package. 
-See [Flutter: read & write files](https://flutter.dev/docs/cookbook/persistence/reading-writing-files) for more info. 
-Have a look how it's done in the Flutter example app:
 ```dart
 import 'package:path_provider/path_provider.dart';
 
+import 'objectbox.g.dart'; // created by `pub run build_runner build`
+
 class _MyHomePageState extends State<MyHomePage> {
   Store _store;
 
   @override
   void initState() {
     super.initState();
-    
-    getApplicationDocumentsDirectory().then((dir) {
-      _store = Store(getObjectBoxModel(), directory: dir.path + "/objectbox");
 
-      // See below as well as examples/flutter/objectbox_demo for examples how to use objectbox
+    getApplicationDocumentsDirectory().then((Directory dir) {
+      // Note: getObjectBoxModel() is generated for you in objectbox.g.dart
+      _store = Store(getObjectBoxModel(), directory: dir.path + '/objectbox');
     });
   }
+
+  @override
+  void dispose() {
+    _store?.close();  // don't forget to close the store
+    super.dispose();
+  }
 }
 ```
 
-Dart native
------------
-In the following file, e.g. `models.dart`, we import objectbox.dart to get definitions for `@Entity`, 
-`@Id` and other annotations and define a single entity that should be persisted by ObjectBox. You could have multiple 
-entities in the same file or you can have them spread across multiple files in the `lib` directory tree. 
+See [Flutter: read & write files](https://flutter.dev/docs/cookbook/persistence/reading-writing-files) for more info.
+If you didn't specify this path to ObjectBox, it would try to use a default directory "objectbox" in the current working
+directory, but it doesn't have permissions to write there: `failed to create store: 10199 Dir does not exist: objectbox (30)`.
+
+### Dart CLI apps
+
+In standard apps (not on mobile), ObjectBox can write anywhere the current user can. The following minimal example
+omits the argument to `Store(directory: )`, thus using the default - 'objectbox' in the current working directory.
 
 ```dart
-import "package:objectbox/objectbox.dart";
+import 'objectbox.g.dart'; // created by `pub run build_runner build`
```

Comment:
Hm, above the command was `(flutter|dart) pub build_runner build`, am I missing something?

Reply:
```suggestion
import 'objectbox.g.dart'; // created by `dart pub run build_runner build`
```